spec/glueby/fee_provider_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+RSpec.describe 'Glueby::FeeProvider' do
+  let(:wallet_adapter) { double(:wallet_adapter) }
+
+  before do
+    Glueby::Internal::Wallet.wallet_adapter = wallet_adapter
+    allow(wallet_adapter).to receive(:load_wallet)
+  end
+
+  after { Glueby::Internal::Wallet.wallet_adapter = nil }
+
+  describe '#provide' do
+    subject { Glueby::FeeProvider.instance.provide(tx) }
+
+    before do
+      allow(wallet_adapter).to receive(:list_unspent).and_return([utxo_for_paying_fee])
+    end
+
+    let(:utxo_for_paying_fee) do
+      {
+        txid: '5c3d79041ff4974282b8ab72517d2ef15d8b6273cb80a01077145afb3d5e7cc5',
+        script_pubkey: '76a914234113b860822e68f9715d1957af28b8f5117ee288ac',
+        vout: 0,
+        amount: 1000,
+        finalized: true
+      }
+    end
+    # The tx that would be added an input for paying fee by FeeProvider
+    # Initially it has one input and two outputs which is for transfer and change.
+    let(:tx) { Tapyrus::Tx.parse_from_payload("01000000018939a2974964f144941ed79c60a0046ee9f842f3b60453ea7d4533601c7ca030000000006a473044022004e5c255e7f397d3ea67e104b00bebfc31cff8d8f57b4c4972473936dda35d8e0220562676a1d1dfd85c89d596e7e96eaa178f8ef4b30c109c561612435137a8860481210280b5f253e612294a2c15a912bbb39b1d5ea48163c14bff56fd1a7079fed23258ffffffff02e8030000000000001976a91402b5fb1195e4c7b069ac33751d03d73a216babe588ac28230000000000001976a9148981ef639f525dc87b0c92ea9fb9af5148cf229688ac00000000".htb) }
+    let(:added_fee_input_tx) { Tapyrus::Tx.parse_from_payload("01000000028939a2974964f144941ed79c60a0046ee9f842f3b60453ea7d4533601c7ca030000000006a473044022004e5c255e7f397d3ea67e104b00bebfc31cff8d8f57b4c4972473936dda35d8e0220562676a1d1dfd85c89d596e7e96eaa178f8ef4b30c109c561612435137a8860481210280b5f253e612294a2c15a912bbb39b1d5ea48163c14bff56fd1a7079fed23258ffffffffc57c5e3dfb5a147710a080cb73628b5df12e7d5172abb8824297f41f04793d5c0000000000ffffffff02e8030000000000001976a91402b5fb1195e4c7b069ac33751d03d73a216babe588ac28230000000000001976a9148981ef639f525dc87b0c92ea9fb9af5148cf229688ac00000000".htb) }
+
+    it 'calls wallet apdater #sign_tx with a tx that is added a fee input' do
+      expect(wallet_adapter).to receive(:sign_tx).with(
+        Glueby::FeeProvider::WALLET_ID,
+        added_fee_input_tx,
+        [],
+        sighashtype: Tapyrus::SIGHASH_TYPE[:all]
+      )
+      subject
+    end
+  end
```

Comment:
The following cases are missing tests.
* Colored coins are not selected as a fee.
* An exception occurs when the fee is insufficient.